src/frontend/src/session.rs

```diff
@@ -652,6 +653,38 @@ pub struct SessionImpl {
     last_idle_instant: Arc<Mutex<Option<Instant>>>,
 
     cursor_manager: Arc<CursorManager>,
+
+    /// temporary sources for the current session
+    temporary_source_manager: Arc<Mutex<TemporarySourceManager>>,
+}
+
+#[derive(Default, Clone)]
+pub struct TemporarySourceManager {
```

Comment:
May add some documentation for the special purposes and limitation of the current impl.